extensions/oplogPopulator/modules/ConnectorsManager.js

```diff
@@ -238,47 +222,98 @@ class ConnectorsManager {
         }
     }
 
+    /**
+     * Spawns a connector when buckets are configured for it and is not running,
+     * or destroys connector with no buckets configured
+     * @param {Connector} connector connector instance
+     * @returns {Promise<Boolean>} true if connector state changed
+     * @throws {InternalError}
+     */
+    async _spawnOrDestroyConnector(connector) {
```

Comment:
Now that bucket can be allocated to non-running connector(s) (before connector gets spawn), could this have an impact on the design? (e.g. potentially missing some MongoDB changes until the connector gets spawned...?)

Reply:
not really, we anyway have a delay between the "activation of new feature which requires oplog" and the moment when a connector is indeed running (i.e. the time for the oplog populator to setup everything) : it may only increase the delay ever so slightly...

but the immediate benefit is that we avoid running any change stream in Veeam/backup use case, and thus avoid risk of breaking the platform.